chamge/amqp-hub-backend.c

```diff
@@ -887,6 +924,141 @@ chamge_amqp_hub_backend_deactivate (ChamgeHubBackend * hub_backend)
   return CHAMGE_RETURN_OK;
 }
 
+gchar *_get_queue_name (const gchar * request);
+
+/*gchar *
+_get_queue_name (const gchar * request)
+{
+  g_autoptr (JsonParser) parser = json_parser_new ();
+  g_autoptr (GError) error = NULL;
+  const gchar *name = NULL;
+
+  JsonNode *root = NULL;
+  JsonObject *json_object = NULL;
+
+  if (!json_parser_load_from_data (parser, request, strlen (request), &error)) {
+    g_debug ("failed to parse body: %s", error->message);
+    return NULL;
+  }
+
+  root = json_parser_get_root (parser);
+  json_object = json_node_get_object (root);
+  if (json_object_has_member (json_object, "to")) {
+    JsonNode *node = json_object_get_member (json_object, "to");
+    name = json_node_get_string (node);
+    return g_strdup (name);
+  }
+  return NULL;
+}
+*/
+
+static ChamgeReturn
+chamge_amqp_hub_backend_user_command (ChamgeHubBackend *
+    hub_backend, const gchar * cmd, gchar ** out, GError ** error)
+{
+  ChamgeAmqpHubBackend *self = CHAMGE_AMQP_HUB_BACKEND (hub_backend);
+  g_autofree gchar *amqp_uri = NULL;
+  struct amqp_connection_info connection_info;
+  amqp_rpc_reply_t amqp_r;
+  amqp_connection_state_t amqp_conn;
+  amqp_socket_t *amqp_socket = NULL;
+
+  g_autofree gchar *amqp_exchange_name = NULL;
+  gint amqp_channel = 0;
+  ChamgeReturn ret = CHAMGE_RETURN_FAIL;
+
+  amqp_uri = g_settings_get_string (self->settings, "amqp-uri");
+  g_assert_nonnull (amqp_uri);
+
+  if (amqp_parse_url (amqp_uri, &connection_info) != AMQP_STATUS_OK) {
+    g_set_error_literal (error, CHAMGE_BACKEND_ERROR,
+        CHAMGE_BACKEND_ERROR_INVALID_PARAMETER, "url pasring failure");
+    goto out;
+  }
+
+  amqp_channel = g_settings_get_int (self->settings, "amqp-channel");
+  amqp_exchange_name =
+      g_settings_get_string (self->settings, "enroll-exchange-name");
```

Comment:
@justinjoy @hkseo95 Is it correct to use "enroll-exchange-name" here? Isn't it something strictly related to "enroll" command?

Reply:
> @justinjoy @hkseo95 Is it correct to use "enroll-exchange-name" here? Isn't it something strictly related to "enroll" command?

Yes, it is corrent. We wanted  this to change in gsettings. "enroll-exchange-name" is exchange name for enroll.